lib/ansible/modules/network/vyos/vyos_vlan.py

```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2017, Ansible by Red Hat, inc
+#
+# This file is part of Ansible by Red Hat
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
+---
+module: vyos_vlan
+version_added: "2.5"
+author: "Trishna Guha (@trishnaguha)"
+short_description: Manage VLANs on VyOS network devices
+description:
+  - This module provides declarative management of VLANs
+    on VyOS network devices.
+notes:
+  - Tested against VYOS 1.1.7
+options:
+  name:
+    description:
+      - Name of the VLAN.
+  address:
+    description:
+      - Configure Virtual interface address.
+  vlan_id:
+    description:
+      - ID of the VLAN. Range 0-4094.
+    required: true
+  interfaces:
+    description:
+      - List of interfaces that should be associated to the VLAN.
+    required: true
+  delay:
+    description:
+      - Delay the play should wait to check for declarative intent params values.
+    default: 10
+  aggregate:
+    description: List of VLANs definitions.
+  purge:
+    description:
+      - Purge VLANs not defined in the I(aggregate) parameter.
+    default: no
+  state:
+    description:
+      - State of the VLAN configuration.
+    default: present
+    choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: Create vlan
+  vyos_vlan:
+    vlan_id: 100
+    name: vlan-100
+    interfaces: eth1
+    state: present
+
+- name: Add interfaces to VLAN
+  vyos_vlan:
+    vlan_id: 100
+    interfaces:
```

Comment:
Pre-requisite is vlan is already created? If not present, does it create one and any default name assigned to the vlan?

Reply:
@kedarX this is just an example to show how one can add multiple interfaces to vlan. For vyos, the vlan is created at the same time when you add interface.